qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -142,11 +145,10 @@ void OutgoingFromQueue::handle(pn_delivery_t* delivery)
                 if (preAcquires()) {
                     //TODO: handle message-annotations
                     if (pn_disposition_is_undeliverable(pn_delivery_remote(delivery))) {
-                        //treat undeliverable here as rejection
-                        queue->reject(r.cursor);
-                    } else {
-                        queue->release(r.cursor, pn_disposition_is_failed(pn_delivery_remote(delivery)));
+                        undeliverableMessages.add(r.msg.getSequence());
                     }
+
+                    queue->release(r.cursor, pn_disposition_is_failed(pn_delivery_remote(delivery)));
```

Comment:
To be fair I think this reverts to the behavior before your patch, correct? As in this isn't breaking any already held expectations.  Having said that, of course this might have been undesired behavior from the get go.

I'm hesitant to add per-queue configuration as I'm not quite sure how I would specify the with amqp 1.0 (iirc most of the create metadata is 0-10?), but am totally willing to explore those options. The broker-level option seems more digestible to me atm.